File: app/src/main/java/eu/electronicid/customdemo/MainActivity.java
package eu.electronicid.customdemo;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import java.net.MalformedURLException;
import java.net.URL;

import eu.electronicid.sdk.videoid.config.Environment;
import eu.electronicid.sdk.videoid.contract.dto.domain.Process;
import eu.electronicid.sdk.videoid.contract.dto.domain.SimilarityLevel;
import eu.electronicid.sdk.videoid.ui.VideoIDActivity;

public class MainActivity extends AppCompatActivity {

    private static final int VIDEOID_REQUEST_CODE = 100;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button button = (Button) findViewById(R.id.button);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                try {
                    final URL eidEndpoint = new URL("https://etrust-sandbox.electronicid.eu/v2");

                    Intent intent = new Intent(MainActivity.this, VideoActivity.class);
                    intent.putExtra(VideoIDActivity.INTENT_ENVIRONMENT, new Environment(eidEndpoint, "---insert_bearer_token---"));
                    intent.putExtra(VideoIDActivity.INTENT_DOCUMENT_TYPE, 62);
                    intent.putExtra(VideoIDActivity.INTENT_MIN_SIMILARITY, SimilarityLevel.Low);
                    intent.putExtra(VideoIDActivity.INTENT_VIDEO_MODE, Process.Unattended);
                    intent.putExtra(VideoIDActivity.INTENT_RAUTHORITY_ID, "---insert_rauthority_id---");
                    startActivityForResult(intent, VIDEOID_REQUEST_CODE);

                } catch (MalformedURLException e) {
                    e.printStackTrace();
                }
            }
        });
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {

        if (requestCode == VIDEOID_REQUEST_CODE) {

            if (resultCode == RESULT_OK) {

                final String videoId = data.getStringExtra(VideoIDActivity.INTENT_RESULT_OK);

                runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        Toast.makeText(MainActivity.this, "Video OK: " + videoId, Toast.LENGTH_SHORT).show();
                    }
                });

            } else if (resultCode == RESULT_CANCELED) {

                if (data != null) {

                    final String errorId = data.getStringExtra(VideoIDActivity.INTENT_RESULT_ERROR_CODE);
                    final String errorMsg = data.getStringExtra(VideoIDActivity.INTENT_RESULT_ERROR_MESSAGE);

                    runOnUiThread(new Runnable() {
                        @Override
                        public void run() {
                            Toast.makeText(MainActivity.this, "Video ERROR id: " + errorId + ", msg: " + errorMsg, Toast.LENGTH_SHORT).show();
                        }
                    });
                }
            }
        }
    }
}

File: app/src/main/java/eu/electronicid/customdemo/MainActivity.kt
package eu.electronicid.customdemo

import android.app.Activity
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import eu.electronicid.customdemo.custom.CustomVideoIdActivity
import eu.electronicid.customdemo.custom.CustomVideoIdActivity2
import eu.electronicid.sdk.video.config.Environment
import eu.electronicid.sdk.video.ui.VideoActivity
import eu.electronicid.sdk.videoid.VideoIDActivity
import kotlinx.android.synthetic.main.activity_main.*
import java.net.URL

class MainActivity : AppCompatActivity() {

    private val REQUEST_CODE = 1
    private val endpoint = URL("https://etrust-sandbox.electronicid.eu/v2/")

    private enum class TypeCustomView { STYLE, CUSTOM_1, CUSTOM_2 }

    private var typeCustomView = TypeCustomView.CUSTOM_2

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button_start_sample.setOnClickListener {
            startActivityForResult(Intent(this,
                    when (typeCustomView) {
                        TypeCustomView.STYLE -> VideoIDActivity::class.java
                        TypeCustomView.CUSTOM_1 -> CustomVideoIdActivity::class.java
                        TypeCustomView.CUSTOM_2 -> CustomVideoIdActivity2::class.java
                    }
            ).apply {
                putExtra(VideoActivity.INTENT_ENVIRONMENT, Environment(endpoint, "S1T14mx3TGGKvTyzBO4gcAChAMHHYdmS1DGSinJczehAGeZTYB64fsC8yu0kwi5r4jBLMBlb_AkXNV4SUZd5wIIkQzqnSKjPJAyhHBwsCxY="))
                putExtra(VideoActivity.INTENT_LANGUAGE, "en")
                putExtra(VideoIDActivity.INTENT_DOCUMENT_TYPE, 62)
            }, REQUEST_CODE)
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        if (requestCode == REQUEST_CODE) {
            if (resultCode == Activity.RESULT_OK) {
                data?.run {
                    val videoId = getStringExtra(VideoActivity.INTENT_RESULT_OK)

                    Toast.makeText(this@MainActivity, "Video OK: $videoId", Toast.LENGTH_SHORT).show()
                }
            } else if (resultCode == Activity.RESULT_CANCELED) {
                data?.run {
                    val errorId = getStringExtra(VideoActivity.INTENT_RESULT_ERROR_CODE)
                    val errorMsg = getStringExtra(VideoActivity.INTENT_RESULT_ERROR_MESSAGE)

                    Toast.makeText(this@MainActivity, "Video ERROR id: $errorId, msg: $errorMsg", Toast.LENGTH_SHORT).show()
                }
            }
        }
    }
}
